docgen/src/Styling.md

```diff
@@ -10,58 +10,133 @@ All widgets that render one or more DOM nodes accept a `theme` prop. This prop i
 
-```
-const searchBoxDefaultTheme = SearchBox.defaultTheme;
-```
+In this tutorial, you'll learn how to: 
 
-### Example: Styling a SearchBox
+* Customize a default provided theme. 
+* Create a totally new theme for a widget.  
 
-#### Redefining class names 
+Before we start, you need to know that this tutorial will use css-modules as a way to style components. Be aware that you are totally free to use any other way supported by [`react-themeable`](https://github.com/markdalgleish/react-themeable). 
+Please also note that every `react-instantsearch` widgets already have class names and if you want you can use them directly. 
 
-If you want to use pure css on your side, either you can use the generated class names we provide for each widgets, or you can redefine your own. You can also directly use some css. 
+## Modify the SearchBox colors 
 
+The `SearchBox` is the most opinionated widget that you will find in `react-instantsearch`. It's already nice out-of-the-box but the submit button background color may not fit your own theme.
+To change it, let's create a new css file called `searchBox.css`. 
+
+```css
+.submit {
+  background-color: #1E9C5E;
+}
 ```
-import {SearchBox} from 'react-instantsearch';
 
-export default function ThemedSearchBox(props) {
+Now that we have redefine the background color of the submit button to green, we need to import it. But because we don't want to redefine completely the `SearchBox` theme, we are going to
+use the `mergeClassNames` function provided by `react-instantsearch`. 
+
+```js
+//We need to add mergeClassNames to our import
+import {InstantSearch, Hits, SearchBox, hightlight, RefinementList, Pagination, CurrentFilters, mergeClassNames} from 'react-instantsearch';
+
+//we need to import our custom SearchBox style
+import searchBoxStyle from './searchbox.css';
+
+// [...]
+
+function Search() {
   return (
-    <SearchBox
-      {...props}
-      theme={{
-        root: 'MySearchBox',
-        wrapper: 'MySearchBox__wrapper',
-        input: {
-          color: 'red',
-        },
-        submit: 'MySearchBox__submit',
-        reset: {
-          display: 'none',
-        },
-      }}
-    />
+    <div className='container'>
+      <CurrentFilters/>
+      <SearchBox theme={mergeClassNames(SearchBox.defaultClassNames, searchBoxStyle)}/>
+      <RefinementList attributeName="category" />
+      <Hits itemComponent={Product} /> 
+      <Pagination />
+    </div>
   );
 }
 ```
 
-If you want to use css-modules, aphrodite and other libraries supported by react-themeable you will also need to use the `theme` prop. Please refer to the [`react-themeable documentation`](https://github.com/markdalgleish/react-themeable)
-to see the correct syntax. 
+You can retrieve the default class names used by a widget by accessing the `defaultClassNames` attribute on a component. 
+The `mergeClassNames` function will create a new object, containing both default and new class names.
```

Comment:
Maybe we can have a link to the function documentation? (even if it's empty for the moment)